lib/block-patterns.php

```diff
@@ -236,6 +236,58 @@ function load_remote_patterns() {
 	}
 }
 
+/**
+ * Register featured patterns from wordpress.org/patterns.
+ *
+ * @since 5.9.0
+ */
+function load_remote_featured_patterns() {
+	/**
+	 * Filter to disable remote block patterns.
+	 *
+	 * @since 5.8.0
+	 *
+	 * @param bool $should_load_remote
+	 */
+	$should_load_remote = apply_filters( 'should_load_remote_block_patterns', true );
+
+	if ( ! $should_load_remote ) {
+		return;
+	}
+
+	// TODO: check about the `featured` name.. Should this be reserved?
+	// Also check core logic: https://developer.wordpress.org/reference/classes/wp_block_pattern_categories_registry/
+	// Do we just allow collisions/overide with categories when they have the same name? Needs checking..
+	if ( ! WP_Block_Pattern_Categories_Registry::get_instance()->is_registered( 'Featured' ) ) {
+		register_block_pattern_category( 'featured', array( 'label' => __( 'Featured', 'gutenberg' ) ) );
+	}
+
+	$request = new WP_REST_Request( 'GET', '/wp/v2/pattern-directory/patterns' );
+	// TODO: change this with new category id when created in patterns directory..
+	$request->set_param( 'category', 4 ); // temp gallery category id.
```

Comment:
```suggestion
	$request['category'] = 4; // temp gallery category id.
```

We try to stick to using the array notation on the request object.